File: 03. Recursion/factorial.cpp
#include <iostream>
using namespace std;

int factorial(int num){

    if(num==0)
    return 1;
    else
    return num*factorial(num-1);
}//factorial

int main(){
    
    int num=5;

    cout<<"The factorial of "<<num<<" is "<< factorial(num);
    return 0;
}//main

File: 03. Recursion/Factorial.java
class Factorial {

    //Compute the factorial of n through recursion
    //Beware of overflows for large values
    public static int factorial(int n){
        if (n > 0) {
            return n * factorial(n - 1);
        }
        else {
            return 1;
        }
    }

    public static void main(String[] args){
        System.out.println(factorial(15));
    }
}